### PR TITLE
Improve usability

### PR DESCRIPTION
**Pull request by malteserteresa**
- Created a ParseArgs class that acts as a decorator to the model class. This acts like git commands ie. python model <command> (eg. build, predict) [<args>] (eg. verbose, n_words etc)
- improved test coverage (but this still needs improving at)
- moved model into its own class

TODO
- update readme
- working for predict

### CODE REVIEW

File: parseargs.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
+import argparse
+import sys
+
+import pytest
+
+
+class ParseArgs:
```
Comment (cmcaine) [inline parseargs.py]: I think it makes more sense for this to be the CLI.

Don't import or call this from model.py, do it the other way around.

Also looks like the predict subcommand doesn't work. Come up with some way of saving and loading a model (and add them as functions or methods in model.py) and call those methods from here.

This becomes the CLI entry point to training a model.

---

If you write other CLIs in the future, you might want to look up alternatives to argparse that are a bit easier to use. [Docopt](https://github.com/docopt/docopt) is popular. I wrote [cli](https://github.com/cmcaine/cli), which I obviously prefer.

You don't need to change it now, but the normal way to handle subcommands is with "subparsers". I use them in `generate_parser_obj()` in cli.py

Or you could just write something like this:

```python
class StopItCLI:
    def build(path_to_data, text_column_name, label_column_name, *, 
            vocab_size=10000, batch_size=512, ...):
        Model().build(path_to_data, text_column_name, label_column_name,
                vocab_size, ...)

    def predict(sentence):
        pre_fitted_model = whatever
        print(Model().predict(sentence, pre_fitted_model, ...))

import cli
cli(StopItCLI)()
```

File: model.py [hunk 2]
```diff
@@ -6,147 +6,149 @@
 from keras.preprocessing.text import Tokenizer
 from tensorflow import keras
 
+from parseargs import ParseArgs
 from utils import save_embeddings
 from visualization import plot_loss, plot_accuracy
 
-vocab_size = 10000
 
+class Model:
```
Comment (cmcaine) [inline model.py]: I don't really see why this is a class. There's no data associated with each Model instance - it's just a collection of functions.

File: model.py [hunk 3]
```diff
@@ -6,147 +6,149 @@
...
+
+    def __init__(self):
```
Comment (cmcaine) [inline model.py]: You can omit this.

File: model.py (hunk 2 above)
Comment (cmcaine) [reply model.py]: As is, the interface is a bit confusing.

I'd suggest bundling the keras model and corpus vocabulary together - it's not appropriate to use a model with a different corpus than it was trained on anyway (as I understand), so don't let them get separated.

You could do that by expanding the model class to contain all the relevant data.

Comment (cmcaine) [reply model.py]: Anyway, then the interface might look like this:

```python
model = build_model(lots of parameters)
save_model(model, 'filename')

# later
model = load_model('filename')

# In use:
model.predict('sentence')
```

build/save/load_model could be static methods of the model class or just module functions.

**Comment by cmcaine** [comment]
Review above as requested ^^

File: model.py (hunk 2 above)
Comment (malteserteresa) [reply model.py]: Here what you're saying, I did this so the interface is enforced whenever another class wants to interact with the anything to do with the model. Shall rethink this in the future

File: parseargs.py (hunk 1 above)
Comment (malteserteresa) [reply parseargs.py]: Will go with what we've got for now but look at this for next time!

**Pull request merged**